packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -105,12 +111,41 @@ export const treeNodeApi: TreeNodeApi = {
 		// The parent of `node` is an object, a map, or undefined (and therefore `node` is a root/detached node).
 		return parentField.parent.key;
 	},
-	on: <K extends keyof FlexTreeNodeEvents>(
+	on: <K extends keyof TreeChangeEvents>(
 		node: TreeNode,
 		eventName: K,
-		listener: FlexTreeNodeEvents[K],
+		listener: TreeChangeEvents[K],
 	) => {
-		return getFlexNode(node).on(eventName, listener);
+		const flex = getFlexNode(node);
+		const anchor = flex.anchorNode;
+
+		switch (eventName) {
+			case "afterShallowChange": {
+				// The funky pattern subscribing to two events from the anchors is so we can fire afterShallowChange once,
+				// batching changes to several fields of the node. 'childrenChanged' on the anchor fires on every change to
+				// a field so doesn't allow us to batch.
+				let shouldFireShallowChange = false;
+				const unsubscribeFromChildrenChanged = anchor.on(
+					"childrenChanged",
+					() => (shouldFireShallowChange = true),
```

Comment:
Is this different from `() => { shouldFireShallowChange = true; }` in a meaningul way?

Reply:
Nope, not sure if this was a remnant of a previous version or I just typoed the brace into a parens. Updating.